src/test/java/org/sqlite/DBMetaDataTest.java

```diff
@@ -1568,4 +1570,313 @@ public void version() throws Exception {
         assertThat(meta.getDatabaseMinorVersion()).as("db minor version").isEqualTo(minorVersion);
         assertThat(meta.getUserName()).as("user name").isNull();
     }
+
+    @Nested
+    class DBMetadataTestWithAttachedDatabases {
+        File testDB;
+
+        @BeforeEach
+        public void init() throws IOException, SQLException {
+            testDB = Files.createTempFile("temp", ".sqlite").toFile();
+            stat.executeUpdate("attach database \"" + testDB.toURI().toURL() + "\" as db2;");
+            stat.executeUpdate(
+                    "create table db2.test2 (id integer primary key, fn float default 0.0, sn not null, intvalue integer"
+                            + "(5), realvalue real(8,3));");
+            stat.executeUpdate("create view db2.testView2 as select * from db2.test2;");
+        }
+
+        @Test
+        public void readSchemaWithAttachedDatabases() throws SQLException {
+            ResultSet rs = meta.getSchemas();
+            assertThat(rs.next()).isTrue();
+            ResultSetMetaData rsMeta = rs.getMetaData();
+            assertThat(rsMeta.getColumnCount()).isEqualTo(2);
+            assertThat(rsMeta.getColumnName(1)).isEqualTo("TABLE_SCHEM");
+            assertThat(rs.getString("TABLE_SCHEM")).isEqualTo("main");
+            assertThat(rs.next()).isTrue();
+            assertThat(rs.getString("TABLE_SCHEM")).isEqualTo("db2");
+            assertThat(rsMeta.getColumnName(2)).isEqualTo("TABLE_CATALOG");
+        }
+
+        @Test
+        public void getTablesForAttachedDatabase() throws SQLException {
```

Comment:
we should probably have a test where schema is passed as `null` in `getTables`, and verify that returned tables do have the correct `schema` set.